test/framework/bootstrap/kindprovider.go

```diff
@@ -0,0 +1,92 @@
+//go:build e2e
+
+// Copyright 2024 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package bootstrap
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/cluster-api/test/framework/bootstrap"
+)
+
+// CreateKindBootstrapClusterAndLoadImagesInput is the input for CreateKindBootstrapClusterAndLoadImages.
+type CreateKindBootstrapClusterAndLoadImagesInput struct {
+	// NodeImageRepository defines the image repository to use for the bootstrap cluster nodes.
+	NodeImageRepository string
+
+	bootstrap.CreateKindBootstrapClusterAndLoadImagesInput
+}
+
+// CreateKindBootstrapClusterAndLoadImages returns a new Kubernetes cluster with pre-loaded images.
+func CreateKindBootstrapClusterAndLoadImages(
```

Comment:
I see you left out `WithExtraPortMappings`, was that intentional? 
https://github.com/kubernetes-sigs/cluster-api/blob/b5a78b80173372a8f1eca0db746684315b93a8a2/test/framework/bootstrap/kind_util.go#L84

Reply:
That only exists in `main` not in in [1.6 release](https://github.com/kubernetes-sigs/cluster-api/blob/release-1.6/test/framework/bootstrap/kind_util.go#L37-L55.)